// _@ant-design_pro-form@1.0.0-beta.22@@ant-design/pro-form/lib/components/Radio/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

require("antd/es/radio/style");

var _radio = _interopRequireDefault(require("antd/es/radio"));

var _react = _interopRequireDefault(require("react"));

var _BaseForm = require("../../BaseForm");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var RadioGroup = _radio.default.Group;

var Group = function Group(_ref) {
  var children = _ref.children,
      fieldProps = _ref.fieldProps,
      options = _ref.options;

  var renderChildren = function renderChildren() {
    if (options) {
      return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, options.map(function (option) {
        if (typeof option === 'string') {
          return /*#__PURE__*/_react.default.createElement(_radio.default, {
            key: option,
            value: option
          }, option);
        }

        return /*#__PURE__*/_react.default.createElement(_radio.default, {
          disabled: option.disable,
          key: option.value,
          value: option.value
        }, option.label);
      }), children);
    }

    return children;
  };

  return /*#__PURE__*/_react.default.createElement(RadioGroup, fieldProps, renderChildren());
};
/**
 * Radio
 * @param
 */


var ProFormRadio = function ProFormRadio(_ref2) {
  var fieldProps = _ref2.fieldProps;
  return /*#__PURE__*/_react.default.createElement(_radio.default, fieldProps);
}; // @ts-expect-error


var WrappedProFormRadio = (0, _BaseForm.createField)(ProFormRadio, {
  valuePropName: 'checked'
});
WrappedProFormRadio.Group = (0, _BaseForm.createField)(Group, {
  customLightMode: true
});
WrappedProFormRadio.Button = _radio.default.Button;
var _default = WrappedProFormRadio;
exports.default = _default;
