// _@ant-design_pro-form@1.0.0-beta.22@@ant-design/pro-form/lib/components/UploadDragger/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

require("antd/es/upload/style");

var _upload = _interopRequireDefault(require("antd/es/upload"));

var _react = _interopRequireDefault(require("react"));

var _icons = require("@ant-design/icons");

var _BaseForm = require("../../BaseForm");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

/**
 * 拖动上传组件
 * @param
 */
var ProFormUploadDragger = function ProFormUploadDragger(_ref) {
  var fieldProps = _ref.fieldProps,
      _ref$title = _ref.title,
      title = _ref$title === void 0 ? '单击或拖动文件到此区域进行上传' : _ref$title,
      _ref$icon = _ref.icon,
      icon = _ref$icon === void 0 ? /*#__PURE__*/_react.default.createElement(_icons.InboxOutlined, null) : _ref$icon,
      _ref$description = _ref.description,
      description = _ref$description === void 0 ? '支持单次或批量上传' : _ref$description,
      action = _ref.action,
      accept = _ref.accept,
      children = _ref.children;
  return /*#__PURE__*/_react.default.createElement(_upload.default.Dragger, _extends({
    name: "files",
    action: action,
    accept: accept
  }, fieldProps), /*#__PURE__*/_react.default.createElement("p", {
    className: "ant-upload-drag-icon"
  }, icon), /*#__PURE__*/_react.default.createElement("p", {
    className: "ant-upload-text"
  }, title), /*#__PURE__*/_react.default.createElement("p", {
    className: "ant-upload-hint"
  }, description), children);
};

var _default = (0, _BaseForm.createField)(ProFormUploadDragger, {
  valuePropName: 'fileList',
  getValueFromEvent: function getValueFromEvent(value) {
    return value.fileList;
  }
});

exports.default = _default;
