// _@ant-design_pro-form@1.0.0-beta.22@@ant-design/pro-form/es/components/Radio/index.js
import "antd/es/radio/style";
import _Radio from "antd/es/radio";
import React from 'react';
import { createField } from '../../BaseForm';
var RadioGroup = _Radio.Group;

var Group = function Group(_ref) {
  var children = _ref.children,
      fieldProps = _ref.fieldProps,
      options = _ref.options;

  var renderChildren = function renderChildren() {
    if (options) {
      return /*#__PURE__*/React.createElement(React.Fragment, null, options.map(function (option) {
        if (typeof option === 'string') {
          return /*#__PURE__*/React.createElement(_Radio, {
            key: option,
            value: option
          }, option);
        }

        return /*#__PURE__*/React.createElement(_Radio, {
          disabled: option.disable,
          key: option.value,
          value: option.value
        }, option.label);
      }), children);
    }

    return children;
  };

  return /*#__PURE__*/React.createElement(RadioGroup, fieldProps, renderChildren());
};
/**
 * Radio
 * @param
 */


var ProFormRadio = function ProFormRadio(_ref2) {
  var fieldProps = _ref2.fieldProps;
  return /*#__PURE__*/React.createElement(_Radio, fieldProps);
}; // @ts-expect-error


var WrappedProFormRadio = createField(ProFormRadio, {
  valuePropName: 'checked'
});
WrappedProFormRadio.Group = createField(Group, {
  customLightMode: true
});
WrappedProFormRadio.Button = _Radio.Button;
export default WrappedProFormRadio;
