// _@ant-design_pro-form@1.0.0-beta.22@@ant-design/pro-form/es/layouts/StepsForm/StepForm.js
import "antd/es/button/style";
import _Button from "antd/es/button";

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { Promise.resolve(value).then(_next, _throw); } }

function _asyncToGenerator(fn) { return function () { var self = this, args = arguments; return new Promise(function (resolve, reject) { var gen = fn.apply(self, args); function _next(value) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value); } function _throw(err) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err); } _next(undefined); }); }; }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { if (typeof Symbol === "undefined" || !(Symbol.iterator in Object(arr))) return; var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React, { useContext, useRef, useState, useEffect } from 'react';
import BaseForm from '../../BaseForm';
import { StepsFormProvide } from './index';

var isBoolean = function isBoolean(val) {
  return typeof val === 'boolean';
};

var StepForm = function StepForm(_ref) {
  var onFinish = _ref.onFinish,
      step = _ref.step,
      restProps = _objectWithoutProperties(_ref, ["onFinish", "step"]);

  var formRef = useRef();
  var context = useContext(StepsFormProvide);

  var _useState = useState(false),
      _useState2 = _slicedToArray(_useState, 2),
      loading = _useState2[0],
      setLoading = _useState2[1];

  var next = /*#__PURE__*/React.createElement(_Button, {
    key: "next",
    type: "primary",
    loading: loading,
    onClick: /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee() {
      var _formRef$current;

      return regeneratorRuntime.wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              (_formRef$current = formRef.current) === null || _formRef$current === void 0 ? void 0 : _formRef$current.submit();

            case 1:
            case "end":
              return _context.stop();
          }
        }
      }, _callee);
    }))
  }, "\u4E0B\u4E00\u6B65");
  var pre = /*#__PURE__*/React.createElement(_Button, {
    key: "pre",
    onClick: function onClick() {
      context === null || context === void 0 ? void 0 : context.pre();
    }
  }, "\u4E0A\u4E00\u6B65");
  var submit = /*#__PURE__*/React.createElement(_Button, {
    key: "submit",
    type: "primary",
    onClick: /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee2() {
      var _formRef$current2;

      return regeneratorRuntime.wrap(function _callee2$(_context2) {
        while (1) {
          switch (_context2.prev = _context2.next) {
            case 0:
              (_formRef$current2 = formRef.current) === null || _formRef$current2 === void 0 ? void 0 : _formRef$current2.submit();

            case 1:
            case "end":
              return _context2.stop();
          }
        }
      }, _callee2);
    }))
  }, "\u63D0\u4EA4");
  /**
   * dom 不存在的时候解除挂载
   */

  useEffect(function () {
    return function () {
      if (restProps.name) {
        context === null || context === void 0 ? void 0 : context.unRegForm(restProps.name);
      }
    };
  }, []);
  var propsSubmitter = isBoolean(restProps.submitter) ? {} : restProps.submitter;

  var getActionButton = function getActionButton() {
    var index = step || 0;

    if (index < 1) {
      return [next];
    }

    if (index + 1 === (context === null || context === void 0 ? void 0 : context.keyArray.length)) {
      return [pre, submit];
    }

    return [pre, next];
  };

  return /*#__PURE__*/React.createElement(BaseForm, _extends({
    formRef: formRef,
    onFinish: /*#__PURE__*/function () {
      var _ref4 = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee3(values) {
        var success;
        return regeneratorRuntime.wrap(function _callee3$(_context3) {
          while (1) {
            switch (_context3.prev = _context3.next) {
              case 0:
                if (restProps.name) {
                  context === null || context === void 0 ? void 0 : context.onFormFinish(restProps.name, values);
                }

                if (!onFinish) {
                  _context3.next = 15;
                  break;
                }

                setLoading(true);
                _context3.prev = 3;
                _context3.next = 6;
                return onFinish === null || onFinish === void 0 ? void 0 : onFinish(values);

              case 6:
                success = _context3.sent;

                if (success) {
                  context === null || context === void 0 ? void 0 : context.next();
                }

                _context3.next = 13;
                break;

              case 10:
                _context3.prev = 10;
                _context3.t0 = _context3["catch"](3);
                // eslint-disable-next-line no-console
                console.error(_context3.t0);

              case 13:
                setLoading(false);
                return _context3.abrupt("return");

              case 15:
                context === null || context === void 0 ? void 0 : context.next();

              case 16:
              case "end":
                return _context3.stop();
            }
          }
        }, _callee3, null, [[3, 10]]);
      }));

      return function (_x) {
        return _ref4.apply(this, arguments);
      };
    }(),
    layout: "vertical",
    contentRender: function contentRender(items, submitter) {
      return /*#__PURE__*/React.createElement(React.Fragment, null, items, submitter);
    }
  }, restProps, {
    submitter: restProps.submitter === false ? false : _objectSpread(_objectSpread({}, propsSubmitter || {}), {}, {
      // 反转按钮，在正常模式下，按钮应该是主按钮在前
      render: function render(submitterProps) {
        var dom = getActionButton();

        var _ref5 = propsSubmitter || {},
            render = _ref5.render;

        if (render === false) {
          return false;
        }

        if (render) {
          return render(submitterProps, dom);
        }

        return dom;
      }
    })
  }));
};

export default StepForm;
