// _@ant-design_pro-form@1.0.0-beta.22@@ant-design/pro-form/es/components/UploadButton/index.js
import "antd/es/upload/style";
import _Upload from "antd/es/upload";
import "antd/es/button/style";
import _Button from "antd/es/button";

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import { UploadOutlined } from '@ant-design/icons';
import { createField } from '../../BaseForm';
/**
 * 上传按钮组件
 * @param
 */

var ProFormUploadButton = function ProFormUploadButton(_ref, ref) {
  var fieldProps = _ref.fieldProps,
      action = _ref.action,
      accept = _ref.accept,
      listType = _ref.listType,
      _ref$title = _ref.title,
      title = _ref$title === void 0 ? '单击上传' : _ref$title,
      _ref$icon = _ref.icon,
      icon = _ref$icon === void 0 ? /*#__PURE__*/React.createElement(UploadOutlined, null) : _ref$icon;
  return /*#__PURE__*/React.createElement(_Upload, _extends({
    action: action,
    accept: accept,
    ref: ref,
    name: "fileList",
    listType: listType || 'picture'
  }, fieldProps), /*#__PURE__*/React.createElement(_Button, null, icon, title));
};

export default createField(React.forwardRef(ProFormUploadButton), {
  valuePropName: 'files',
  getValueFromEvent: function getValueFromEvent(value) {
    return value.fileList;
  }
});
